### PR TITLE
postgres-cdc: introduce logic to convert state from Debezium 2.1.2 to Debezium 1.9.6

### PR DESCRIPTION
**Pull request by subodh1810**
I branched out of this commit SHA 3843eca10a4a98d3bfc2a98858657ce80142a399 (which is just before I merged my [PR](https://github.com/airbytehq/airbyte/pull/23112) to upgrade Debezium in master) and then made this change. This PR can not be merged in master but should allow us to release an interim Postgres image which can convert the state from Debzium 2.1.2 back to 1.9.6 in case we have to roll back cause of some issues. 

See : https://airbytehq-team.slack.com/archives/C02U46QK5C3/p1678217190006939?thread_ts=1678214731.413079&cid=C02U46QK5C3

https://github.com/airbytehq/airbyte/compare/3843eca10a4a98d3bfc2a98858657ce80142a399...postgres-cdc-state-downgrade-1-9-6

This is how the state from 1.9.6 looks likes in our DB 
```
id            | 0353f52c-84e6-4e7e-9ffc-b7584a467dcb
connection_id | 3430d2af-7725-4cea-a612-0f9b788eb8d2
state         | {"state": {"{\"schema\":null,\"payload\":[\"postgres\",{\"server\":\"postgres\"}]}": "{\"transaction_id\":null,\"lsn_proc\":132087920,\"lsn_commit\":132063120,\"lsn\":132087920,\"txId\":248567,\"ts_usec\":1678223259335261}"}}
created_at    | 2023-03-07 21:04:24.284108+00
updated_at    | 2023-03-07 21:08:02.494463+00
stream_name   | 
namespace     | 
type          | GLOBAL
```

And this is how state from 2.1.2 looks like in our DB 
```
id            | 0353f52c-84e6-4e7e-9ffc-b7584a467dcb
connection_id | 3430d2af-7725-4cea-a612-0f9b788eb8d2
state         | {"state": {"[\"postgres\",{\"server\":\"postgres\"}]": "{\"transaction_id\":null,\"lsn\":132059768,\"txId\":248521,\"ts_usec\":1678223031163372}"}}
created_at    | 2023-03-07 21:04:24.284108+00
updated_at    | 2023-03-07 21:04:24.284108+00
stream_name   | 
namespace     | 
type          | GLOBAL
```

### CODE REVIEW

File: airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/AirbyteFileOffsetBackingStore.java [hunk 1]
```diff
@@ -59,14 +60,40 @@ public Map<String, String> read() {
...
         e -> stringToByteBuffer(e.getKey()),
         e -> stringToByteBuffer(e.getValue())));
 
     FileUtils.deleteQuietly(offsetFilePath.toFile());
     save(mappedAsStrings);
   }
 
+  // This is to allow us to release Debezium 2.1.2 safely so that in case of a roll back we are able
+  // to convert the new state from Debezium 2.1.2 back to the format of Debezium 1.9.6
+  private Map<String, String> downgradeStateForDebezium1_9(final Map<String, String> mapAsString) {
```
Comment (akashkulk) [inline airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/AirbyteFileOffsetBackingStore.java:75]: Should we only downgrade state if it 2.1 (i.e. 1.9 -> 1.9 should be a no-op)

Comment (subodh1810) [reply airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/AirbyteFileOffsetBackingStore.java:75]: Yup thats what its doing. It has the logic in place for that. 
```
if (i == 0 && i1 == key.length() - 1) {
        // The state is Debezium 1.9 compatible. No need to change anything.
        return mapAsString;
      }
```

Comment (subodh1810) [reply airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/AirbyteFileOffsetBackingStore.java:75]: I need to extract the keys to make that decision

Comment (akashkulk) [reply airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/AirbyteFileOffsetBackingStore.java:75]: Wow, sorry missed this. Could you (maybe in the PR description) give an example of what the 1.9 state looks like vs 2.1 state? Just thinking of a case where it's easy for someone oncall (me..) to go back and easily determine whether a state is assc with 1.9 or 2.1

**Review by akashkulk — COMMENTED** [review]
Bump Dockerfile too?

File: airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/AirbyteFileOffsetBackingStore.java (hunk 1 above)
Comment (subodh1810) [reply airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/AirbyteFileOffsetBackingStore.java:75]: Done

**Comment by subodh1810** [comment]
I bump docker file after approval

> Bump Dockerfile too?

**Review by akashkulk — APPROVED** [review]

**Comment by prateekmukhedkar** [comment]
@subodh1810 do we need to release these changes in MSSQL and MySQL source connectors also?

**Comment by subodh1810** [comment]
@prateekmukhedkar nope. MySQL is already running in cloud and worked without any trouble. See this https://docs.google.com/document/d/1jfwbslOUGr7hAVE9R3iCzHGq0wBdjvS1XIGqXtEpCxs/edit?usp=sharing

For MSSQL there are no active CDC connectors running in cloud

**Pull request closed without merging**